pkg/reconciler/instances/istio/bootstrap.go

```diff
@@ -24,15 +24,15 @@ func istioPerformerCreator(istioProxyReset proxy.IstioProxyReset, provider clien
 
 	res := func(logger *zap.SugaredLogger) (actions.IstioPerformer, error) {
 		pathsConfig := os.Getenv(istioctlBinaryPathEnvKey)
-		istioctlPaths, err := parsePaths(pathsConfig, validatePath)
+		istioctlPaths, err := parsePaths(pathsConfig, ensureFileExecutable, logger)
```

Comment:
I suggest not mix the `parsePaths` and `ensureFileExecutable` together to follow single responsibility. We can iterate the returned `istioctlPaths` to check `ensureFileExecutable`, then we don't need to pass `isValid func(string) error` as an function to `parsePaths`, and it's also easier to write test